lib/hanami/application/settings.rb

```diff
@@ -1,22 +1,94 @@
 # frozen_string_literal: true
 
+require "dry/configurable"
 require "dry/core/constants"
-require_relative "settings/definition"
-require_relative "settings/struct"
 
 module Hanami
   class Application
     # Application settings
     #
+    # Users are expected to inherit from this class to define their application
+    # settings.
+    #
+    # @example
+    #   # config/settings.rb
+    #   # frozen_string_literal: true
+    #
+    #   require "hanami/application/settings"
+    #   require "my_app/types"
+    #
+    #   module MyApp
+    #     class Settings < Hanami::Application::Settings
+    #       setting :database_url
+    #       setting :feature_flag, default: false, constructor: Types::Params::Bool
+    #     end
+    #   end
+    #
+    # Settings are defined with
+    # [dry-configurable](https://dry-rb.org/gems/dry-configurable/), so you can
```

Comment:
@waiting-for-dev Should we link the exact version that we support?

Reply:
I thought about it, but I reckoned that it would be something with many possibilities to get outdated at some point. Users can look at the gemspec to see the version we are using, which could be a range BTW, and it should be our responsibility to keep dependencies updated anyway. However, if you still prefer it, I'll add it here. Just confirm me your decision :slightly_smiling_face: